src/patternfly/components/Button/button.scss

```diff
@@ -150,7 +147,7 @@
 
   // Add spacing to icon if it appears at the beginning of button
   > :first-child {
-    margin-right: var(--pf-c-button--item--MarginRight);
+    margin-right: var(--pf-c-button--first-child--MarginRight);
```

Comment:
This will add it no matter whether it's an icon or otherwise. Are we really trying to make sure whatever is inside a button is spaced apart slightly? So maybe this is really :not(:last-child)?

Reply:
I think using `:nth-child` like this is dangerous in general. You never know if someone would use something like

```
<button>
  <i class="icon">x</i>
  <span>foo</span>
  <span>bar</span>
</button>
```

in which case, they would likely only want space between "x" and "foo". I think if we want to update this, to do it properly it should probably a BEM element like

```
.pf-c-button__icon {
  margin-right: var(--pf-c-button__icon--MarginRight);
}

<button>
  <i class="pf-c-button__icon icon">x</i>
  <span>foo</span>
  <span>bar</span>
</button>
```

What do you think?

Reply:
Glad you mentioned this. Looks like we have this CSS to undo this `:first-child` CSS in cases where we don't want it, too. We could get rid of that if the thing that added the space was a BEM element.

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/InputGroup/input-group.scss#L73

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Button/button.scss#L320

Reply:
I like having the BEM elements. Makes it more explicit what you are aiming for and less specificity.

Reply:
Created a hbs partial to create an icon that takes an argument for the font awesome class to support this. Opened an issue to update the rest of our buttons to use that partial https://github.com/patternfly/patternfly-next/issues/927